### ISSUE TITLE
Got low confidence scores for single class detection

### ISSUE THREAD

**Issue by yiyingsheng07** [facebookresearch/detr#522]
Hi, I used DETR for single class detection on my own dataset, but I got some problems when doing predictions. All scores for the detected object from 20 queries are quite low (~-13) while the "no-object" class has the hightest score (~4). So after Softmax, the confidence score for the object comes to ~4e-08. However, the location of bboxes looks right. I was wondering why I got low confidence scores? Should I try to change ```<eos_coef>```?

Thanks in advance!

**Comment by AlessandroMagliola** [comment]
I have the same problem.  I built my own dataset and implemented a single class object detection, trained 80 epochs and scores are very very low.
Moreover, I am using the following version:  torch=1.12.0, transformers=4.20.1, cuda= 11.1
